### PR TITLE
New training

### PR DESCRIPTION
**Pull request by treykris**
Links
-----
- Web repo PR: https://github.com/UPchieve/web/pull/504

Description
-----------
- Certifications overhaul
- Training courses

Developer self-review checklist
-------------------------------
- [ ] Potentially confusing code has been explained with comments
- [ ] No warnings or errors have been introduced; all known error cases have been handled
- [ ] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [ ] All edge cases have been addressed

### CODE REVIEW

File: models/Question.js [hunk 1]
```diff
@@ -131,7 +131,8 @@ questionSchema.statics.getSubcategories = function(category) {
       'waves and sound',
       'refraction and reflection',
       'gravity/gen relativity'
-    ]
+    ],
+    upchieve101: ['ethics', 'students']
```
Comment (jkeat) [inline models/Question.js:135]: Note to self: update this

File: controllers/TrainingCtrl.js [hunk 2]
```diff
@@ -118,29 +76,44 @@ module.exports = {
...
+        user.certifications
       )
-      Object.assign(userUpdates, integratedMathUpdate)
 
-      // an onboarded volunteer must have updated their availability and obtained at least one certification
-      if (!user.isOnboarded && user.availabilityLastModifiedAt) {
+      // Create a user action for every subject unlocked
+      for (const subject of unlockedSubjects) {
+        // @note: user.certifications is modified in this.getUnlockedSubjects
+        const hasPassedQuizUserAction =
```
Comment (jkeat) [inline controllers/TrainingCtrl.js:96]: delete to get unlock actions for everything ⭐

**Comment by jkeat** [comment]
todo: ordering of dbutil scripts?

**Comment by treykris** [comment]
Migration order:
1. npx ts-node dbutils/add-training-courses.ts
2. npx ts-node dbutils/add-new-quiz-certs.ts
3. npx ts-node dbutils/add-subjects.ts

Downgrade order:
1. DOWNGRADE=true npx ts-node dbutils/add-new-quiz-certs.ts
2. DOWNGRADE=true npx ts-node dbutils/add-subjects.ts
3. DOWNGRADE=true npx ts-node dbutils/add-training-courses.ts

**Review by jkeat — APPROVED** [review]
looks pretty good

**Pull request merged**